demo/realtime-img2img/main.py

```diff
@@ -0,0 +1,201 @@
+            except WebSocketDisconnect as e:
+                logging.error(f"WebSocket Error: {e}, {user_id}")
+                traceback.print_exc()
+            finally:
+                print(f"User disconnected: {user_id}")
+                user_data.delete_user(user_id)
+
+        async def handle_websocket_data(user_id: uuid.UUID, websocket: WebSocket):
+            if not user_data.check_user(user_id):
+                return HTTPException(status_code=404, detail="User not found")
+            last_time = time.time()
+            try:
+                while True:
+                    if args.timeout > 0 and time.time() - last_time > args.timeout:
+                        await websocket.send_json(
+                            {
+                                "status": "timeout",
+                                "message": "Your session has ended",
+                                "userId": str(user_id),
+                            }
+                        )
+                        await websocket.close()
+                        return
+                    data = await websocket.receive_json()
+                    if data["status"] != "next_frame":
+                        asyncio.sleep(THROTTLE)
+                        continue
+
+                    params = await websocket.receive_json()
+                    params = pipeline.InputParams(**params)
+                    info = pipeline.Info()
+                    params = SimpleNamespace(**params.dict())
+                    if info.input_mode == "image":
+                        image_data = await websocket.receive_bytes()
+                        if len(image_data) == 0:
+                            await websocket.send_json({"status": "send_frame"})
+                            continue
+                        params.image = bytes_to_pil(image_data)
+                    await user_data.update_data(user_id, params)
+                    await websocket.send_json({"status": "wait"})
+                    await asyncio.sleep(THROTTLE)
+
+            except Exception as e:
+                logging.error(f"Error: {e}")
+                traceback.print_exc()
+
+        @app.get("/api/queue")
+        async def get_queue_size():
+            queue_size = user_data.get_user_count()
+            return JSONResponse({"queue_size": queue_size})
+
+        @app.get("/api/stream/{user_id}")
+        async def stream(user_id: uuid.UUID, request: Request):
+            try:
+
+                async def generate():
+                    websocket = user_data.get_websocket(user_id)
+                    last_params = SimpleNamespace()
+                    while True:
+                        last_time = time.time()
+                        params = await user_data.get_latest_data(user_id)
+                        if not vars(params) or params.__dict__ == last_params.__dict__:
+                            await websocket.send_json({"status": "send_frame"})
+                            continue
```

Comment:
minor nit: might be a good idea to `await asyncio.sleep(THROTTLE)` before these two `continue` statements, to avoid spamming a slow client